### ISSUE TITLE
Typing a retrieved Field Schema using yup.Reach()

### ISSUE THREAD

**Issue by lukasdruskis** [jquense/yup#2083]
I've finally decided to upgrade from  0.29 to 1.2 and faced some issues with it's types. I'm Looking for help to find a best solution for typing yup schemas.

I have components and functions, where I have an argument - a field schema (string/number/object, etc)

```ts
interface MyType {
  id: number;
  name: string;
}

const mySchema: ObjectSchema<MyType> = yup
  .object({
    id: yup.number().required(),
    name: yup.string().required(),
  })
  .required();

const idSchema = yup.reach(mySchema, "id");
const nameSchema = yup.reach(mySchema, "name");

function validateField(fieldSchema: Schema) {
  fieldSchema.validate("TEST");
}

validateField(idSchema); // <--- Here compiler complains, where in 0.29 it didn't
```

![image](https://github.com/jquense/yup/assets/38437943/953d160e-4d28-40e9-9e9f-4f14f1ca592a)

It seems that I got used to use `Schema` type everywhere, and now it doesn't satisfy neither Object schema value, nor any Field Schema.

**Comment by lukasdruskis** [comment]
@jquense Wouldn't it make sense to export the Reference type? [Someone provided a workaround for this issue](https://stackoverflow.com/questions/76881812/yup-typing-a-retrieved-field-schema-from-an-object-schema/76903812#76903812)

**Comment by jquense** [comment]
yeah we can export it, want to send a PR?

**Comment by lukasdruskis** [comment]
solved in #2091
